planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -123,14 +105,54 @@ private int getTravelDurationInMinutes(Duration duration) {
                 items.add(item);
 
                 // Update the next event's start time
-                if (i != events.size() - 1) { 
-                    start += event.getDurationInMinutes() + getTravelDurationInMinutes(travelTimeGraph[i][i+1]);
-                }
+                start += event.getDurationInMinutes() + getTravelDurationInMinutes(directionsLegs[i].duration);
             }else {
-                errorMessage = "Sorry, you have too many events in a day!";
-                return Arrays.asList();
+                throw new ItineraryException("Sorry, you have too many events in a day! Try removing some events or shorten their duration");
             }
         }
         return items;
     }
+
+    // Function that makes a request to the Directions API and return a direction route
+    // A DirectionsRoute has field legs, which is an array of DirectionsLegs
+    // Each DirectionsLeg contains starting address, ending address, and duration
+    // DirectionRoute also has int[] waypointOrder, which will be empty if optimized is set to false,
+    // and will contain the new order of waypoints if optimized is set to true
+    // For the MVP, real time traffic is NOT used.
+    private DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) throws ItineraryException{
+        String GoogleApiKey = Keys.GOOGLE_MAPS_API_KEY;
```

Comment:
I don't think this variable is necessary. How about calling |Keys.GOOGLE_MAPS_API_KEY| directly below?

Reply:
You're right! It's fixed now